pkg/proxy/iptables/proxier.go

```diff
@@ -328,6 +333,7 @@ func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
 	// Remove services missing from the update.
 	for name := range proxier.serviceMap {
 		if !activeServices[name] {
+			proxier.recorder.Eventf(proxier.nodeRef, "WatchForServiceFailed", "Watch failed, now stopping service %q", name)
```

Comment:
This case has nothing to do with watch failing - this is a case of the service having been removed.  So every time a user says 'kubectl delete service foo' this will cause two events (because the endpoints hunk is the same) from every node in the cluster.  This doesn't seem very useful to me - what are we trying to do?